src/mbgl/tile/raster_dem_tile.cpp

```diff
@@ -0,0 +1,123 @@
+#include <mbgl/renderer/buckets/hillshade_bucket.hpp>
+#include <mbgl/actor/scheduler.hpp>
+
+namespace mbgl {
+
+RasterDEMTile::RasterDEMTile(const OverscaledTileID& id_,
+                       const TileParameters& parameters,
+                       const Tileset& tileset)
+    : Tile(id_),
+      loader(*this, id_, parameters, tileset),
+      mailbox(std::make_shared<Mailbox>(*Scheduler::GetCurrent())),
+      worker(parameters.workerScheduler,
+             ActorRef<RasterDEMTile>(*this, mailbox)) {
+
+    if ( id.canonical.y == 0 ){
+        // this tile doesn't have upper neighboring tiles so marked those as backfilled
+        neighboringTiles = neighboringTiles | DEMTileNeighbors::NoUpper;
+    }
+
+    if (id.canonical.y + 1 == std::pow(2, id.canonical.z)){
+        // this tile doesn't have lower neighboring tiles so marked those as backfilled
+        neighboringTiles = neighboringTiles | DEMTileNeighbors::NoLower;
+    }
+}
+
+RasterDEMTile::~RasterDEMTile() = default;
+
+void RasterDEMTile::setError(std::exception_ptr err) {
+    loaded = true;
+    observer->onTileError(*this, err);
+}
+
+void RasterDEMTile::setMetadata(optional<Timestamp> modified_, optional<Timestamp> expires_) {
+    modified = modified_;
+    expires = expires_;
+}
+
+void RasterDEMTile::setData(std::shared_ptr<const std::string> data) {
+    pending = true;
+    ++correlationID;
+    worker.invoke(&RasterDEMTileWorker::parse, data, correlationID);
+}
+
+void RasterDEMTile::onParsed(std::unique_ptr<HillshadeBucket> result, const uint64_t resultCorrelationID) {
+    bucket = std::move(result);
+    loaded = true;
+    if (resultCorrelationID == correlationID) {
+        pending = false;
+    }
+    renderable = bucket ? true : false;
+    observer->onTileChanged(*this);
+}
+
+void RasterDEMTile::onError(std::exception_ptr err, const uint64_t resultCorrelationID) {
+    loaded = true;
+    if (resultCorrelationID == correlationID) {
+        pending = false;
+    }
+    observer->onTileError(*this, err);
+}
+
+void RasterDEMTile::upload(gl::Context& context) {
+    if (bucket) {
+        bucket->upload(context);
+    }
+}
+
+
+Bucket* RasterDEMTile::getBucket(const style::Layer::Impl&) const {
+    return bucket.get();
+}
+
+HillshadeBucket* RasterDEMTile::getBucket() const {
+    return bucket.get();
+}
+
+void RasterDEMTile::backfillBorder(const RasterDEMTile& borderTile, const DEMTileNeighbors mask) {
+    int32_t dx = borderTile.id.canonical.x - id.canonical.x;
+    const int8_t dy = borderTile.id.canonical.y - id.canonical.y;
+    const uint32_t dim = pow(2, id.canonical.z);
+    if (dx == 0 && dy == 0) return;
+    if (std::abs(dy) > 1) return;
+    // neighbor is in another world wrap
+    if (std::abs(dx) > 1) {
+        if (std::abs(int(dx + dim)) == 1) {
+            dx += dim;
+        } else if (std::abs(int(dx - dim)) == 1) {
+            dx -= dim;
+        }
+    }
+    const HillshadeBucket* borderBucket = borderTile.getBucket();
```

Comment:
`getBucket()` can return `nullptr` if we have received a tile that is invalid or empty. In this case, the worker code eventually calls `RasterDEMTile::onError` or `RasterDEMTile::onParsed` respectively. The former doesn't explicitly set the `bucket` pointer, which means that it'll remain empty. The latter explicitly sets a `nullptr` (via `std::move`).

We should guard accessing this code, and just don't backfill for this particular tile that is missing.